src/databricks/labs/ucx/aws/credentials.py

```diff
@@ -230,3 +230,37 @@ def run(self, prompts: Prompts, *, single_role=False, role_name="UC_ROLE", polic
 
         self._resource_permissions.create_uc_roles(iam_list)
         self._resource_permissions.save_uc_compatible_roles()
+
+    def delete_uc_roles(self, prompts: Prompts) -> None:
+        uc_roles = self._resource_permissions.load_uc_compatible_roles()
+        if len(uc_roles) == 0:
+            self._resource_permissions.save_uc_compatible_roles()
+            uc_roles = self._resource_permissions.load_uc_compatible_roles()
+        storage_credentials = self._ws.storage_credentials.list()
+
+        uc_role_mapping = {role.role_arn: role.role_name for role in uc_roles}
+        selected_roles = prompts.multiple_choice_from_dict("Select the list of roles created by UCX", uc_role_mapping)
+        if len(selected_roles) == 0:
+            logger.info("No roles selected...")
+            return
+        matching_credentials = [
+            storage_credential
```

Comment:
this list comprehension doesn't fit one line, which is proven to be more difficult to maintain. can you unfold it to a `for` loop?